src/client.js

```diff
@@ -33,33 +59,50 @@ const Client = class Client extends EventEmitter {
     this.mqtt.publish(topic, JSON.stringify(message));
   }
 
-  _connected(connack) {
-    super.emit('connect', connack);
-    this.mqtt.subscribe(['+/devices/+/activations', '+/devices/+/up']);
+  _connect(connack) {
+    this.ee.emit('connect', connack);
+  }
+
+  _error(err) {
+    this.ee.emit('error', err);
   }
 
   _handleMessage(topic, message) {
     var parts = topic.split('/');
+    if (parts[3] !== 'activations' && parts[3] !== 'up') {
+      return;
+    }
     var devId = parts[2];
     var payload = JSON.parse(message.toString());
-    switch (parts[3]) {
-      case 'activations':
-        super.emit('activation', Object.assign({
-          dev_id: devId,
-          app_id: this.appId
-        }, payload));
-        break;
-      case 'up':
-        super.emit('message', Object.assign({
-          dev_id: devId,
-          app_id: this.appId
-        }, payload));
-        break;
+    if (parts.length === 4) { // inject app_id and dev_id, but not for field topics
+      payload.app_id = this.appId;
+      payload.dev_id = devId;
     }
+    this.ee.emit(topic, devId, payload); // full topic, including field if any
+    this.ee.emit(parts.slice(0, 2).concat('+', parts.slice(3)).join('/'), devId, payload); // any device
```

Comment:
They don't, as you can see in `on()` and `off()` we translate their arguments to the topic that we only internally use as the eventName so that the above 2 emitted events reach the right listeners.